components/sic_image_engine/src/operations/horizontal_gradient.rs

```diff
@@ -0,0 +1,73 @@
+use crate::errors::SicImageEngineError;
+use crate::operations::ImageOperation;
+use crate::wrapper::gradient_input::GradientInput;
+use rayon::iter::{IntoParallelRefMutIterator, ParallelIterator};
+use sic_core::{
+    image::{self, DynamicImage, Frame, Pixel},
+    SicImage,
+};
+
+#[derive(Clone, Debug, Copy, PartialEq)]
+pub struct HorizontalGradient {
+    pub colors: GradientInput,
+}
+
+impl HorizontalGradient {
+    pub fn new(colors: GradientInput) -> Self {
+        Self { colors }
+    }
+}
+
+impl ImageOperation for HorizontalGradient {
+    fn apply_operation(&self, image: &mut SicImage) -> Result<(), SicImageEngineError> {
+        match image {
+            SicImage::Static(image) => gradient_static_image(image, self.colors),
+            SicImage::Animated(image) => gradient_animated_image(image.frames_mut(), self.colors),
+        }
+
+        Ok(())
+    }
+}
+
+fn gradient_static_image(img: &mut DynamicImage, input: GradientInput) {
+    let (left_color, right_color) = &input.colors();
+    let (width, height) = img.to_rgba8().dimensions();
+    let mut gradient_dynamic = DynamicImage::new_rgba8(width, height);
+    image::imageops::horizontal_gradient(&mut gradient_dynamic, left_color, right_color);
+    blend_static_image(img, gradient_dynamic);
+}
+
+fn gradient_animated_image(frames: &mut [image::Frame], input: GradientInput) {
+    let (left_color, right_color) = &input.colors();
+    let (width, height) = DynamicImage::ImageRgba8(frames[0].buffer().clone())
+        .to_rgba8()
+        .dimensions();
```

Comment:
Similar to the static image operation, we only need read access for `dimensions`:

```rust
use sic_core::image::GenericImageView;
let (width, height) = frames[0].buffer().dimensions();
```

This saves us from copying the underlying buffer twice :smile: :
```rust
DynamicImage::ImageRgba8(frames[0]
			.buffer()
			.clone() # first copy
		)
        .to_rgba8()  # second copy
        .dimensions();
```